src/lib/services-manager/clients/services_manager.rb

```diff
@@ -209,7 +225,7 @@ def buttons
           PushButton(Id(Id::SHOW_DETAILS), _('Show &Details'))
         ]
 
-        if journal_installed?
+        if journal_loaded?
```

Comment:
You do not need to remember whether the `y2journal` has been loaded or not, you can simply check whether the expected class is defined.

We have a similar problem in installation where we have to handle possibly missing `yast2-registration` (it is present in SLE but missing in openSUSE). We use `unless defined?(::Registration::UrlHelpers)` condition, see [here](https://github.com/yast/yast-installation/blob/af009416be742d108b4849db1c260935af1ba6a7/src/lib/installation/update_repositories_finder.rb#L85).

Reply:
yes, if you do loading of journal in initialize and not lazy load, then you can check existence of Y2Journal itself like `unless defined?(::Y2Journal)`